drivers/docker/driver.go

```diff
@@ -872,8 +872,8 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
 	hostConfig.NetworkMode = driverConfig.NetworkMode
 	if hostConfig.NetworkMode == "" {
 		// docker default
-		logger.Debug("networking mode not specified; using default", "network_mode", defaultNetworkMode)
-		hostConfig.NetworkMode = defaultNetworkMode
+		logger.Debug("networking mode not specified; using default")
+		hostConfig.NetworkMode = "default"
```

Comment:
Hm, I wonder why we did not use "default" from the beginning? Is this a "new" alias? LGTM, but I will also check with the team to see if anyone has any backward compatibility concerns.

Reply:
I am not sure why,
but I tested it on windows in windows container mode and linux container mode
I tested it on mac and Linux, the default the networking mode changes PR os by docker so this check is redundant imo.

Reply:
Looks like we reference NetworkMode if some other parameters are used. Have you tested that these parameters (eg `ipv4_address`) still work when the NetworkMode is `default` instead of `bridge` or `nat`?

Reply:
Code that uses NetworkMode: https://github.com/hashicorp/nomad/blob/2ae6a6faf3746508e23980719ac025f87b10cc04/drivers/docker/driver.go#L964-L985

Reply:
on mac and linux:

ipv4_address and ipv6_address can be used with only user defined networks, attempting to use it with a bridge, host, nat or default will result in error from docker
```   2019-08-01T11:59:29.646+0300 [ERROR] client.alloc_runner.task_runner: running driver failed: alloc_id=46e9fe8a-6d4a-e634-906b-2004ff066140 task=sleepTask error="Failed to start container af3a0d663252a60ae2c7c619224fba1b265bd5671e399d62535e2b889a7a0296: API error (400): user specified IP address is supported on user defined networks only"```
so this change does alter the behavior of the code but it doesn't break any functionality

sample job:
```
job "job51" {
  datacenters = ["dc1"]

  type = "batch"

  group "epictaskgroup1" {
    task "sleepTask" {
      driver = "docker"

      config {
        image = "alpine"
        command = "sleep"
        args = ["60"]
        ipv4_address = "172.17.0.5"
      }
    }
  }
}
```

on windows:
ipv4_address works with default network

samplejob:
```job "job52" {
  datacenters = ["dc1"]

  type = "batch"

  group "epictaskgroup1" {
    task "sleepTask" {
      driver = "docker"

      config {
        image = "mcr.microsoft.com/windows/servercore:1903"
        command = "ping"
        args = ["127.0.0.1", "-n", "60"]
        ipv4_address = "172.18.80.5"
      }

      kill_timeout = "10000s"
    }
  }
}
```
docker inspect
```
"NetworkSettings": {
    ...
    "Networks": {
        "nat": {
            "IPAMConfig": {
                "IPv4Address": "172.18.80.5"
            },
            "Links": null,
            "Aliases": null,
            "NetworkID": "24e56e046fed4fee0b4883784c9f769748573dfd89bdb9abdae67c9fe0e73492",
            "EndpointID": "18cb16c91235bae0043cc59d43abf5c1963fb26a26887128d743f4f73bb05937",
            "Gateway": "172.18.80.1",
            "IPAddress": "172.18.80.5",
            "IPPrefixLen": 20,
            "IPv6Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "MacAddress": "00:15:5d:55:db:e5",
            "DriverOpts": null
        }
    }
}
```

Reply:
Fantastic work. Thanks @ilyaiqoqo